tests/integration/reservations.test.ts

```diff
@@ -88,8 +86,142 @@ describe("POST /reservations", () => {
           value: expect.any(Number) 
         }),
         paymentId: expect.any(Number),
-        room: null,
-        roomId: null,
+        room: expect.objectContaining({ 
+          availableBeds: expect.any(Number), 
+          capacity: expect.any(Number), 
+          hotel: null, 
+          id: expect.any(Number), 
+          number: expect.any(String)
+        }),
+        roomId: expect.any(Number),
+        ticket: expect.objectContaining({ 
+          hasHotel: expect.any(Boolean), 
+          id: expect.any(Number), 
+          isPresencial: expect.any(Boolean), 
+          reservationId: expect.any(Number) 
+        }),
+        userId: expect.any(Number),
+        ticketId: expect.any(Number),
+        user: expect.objectContaining({ 
+          createdAt: expect.any(String), 
+          email: expect.any(String), 
+          id: expect.any(Number), 
+          password: expect.any(String) 
+        }),
+      }),
+    );
+  });
+
+  it("should return 403 when the avaibleBeds are sold out", async () => {
+    const { user, session, ticket, payment } = await reservationRequirementsFactory();
+    const room = await createRoom(1, 0);
+    await createReservation(user.id, ticket.id, payment.id);
+
+    const response = await agent.post("/reservation/room").send({
+      roomId: room.id,
+    }).set("Authorization", "Bearer " + session.token);
+
+    expect(response.status).toEqual(403);
+  });
+
+  it("should return 403 when the user already has a reservation", async () => {
+    const { user, session, ticket, payment } = await reservationRequirementsFactory();
+    const room = await createRoom(1, 1);
+    await createReservation(user.id, ticket.id, payment.id);
+
+    await agent.post("/reservation/room").send({ roomId: room.id }).set("Authorization", "Bearer " + session.token);
+
+    const response = await agent.post("/reservation/room").send({
+      roomId: room.id,
+    }).set("Authorization", "Bearer " + session.token);
+
+    expect(response.status).toEqual(403);
+  });
+
+  it("must switch rooms if the user sends the optional argument of the room to be switched", async () => {
+    const { user, session, ticket, payment } = await reservationRequirementsFactory();
+    const room1 = await createRoom(1, 1);
+    const room2 = await createRoom(1, 1);
+    await createReservation(user.id, ticket.id, payment.id);
+
+    await agent.post("/reservation").send({ roomId: room1.id }).set("Authorization", "Bearer " + session.token);
+
+    const response = await agent.post("/reservation/room").send({
+      roomId: room2.id, changeRoom: room1.id
+    }).set("Authorization", "Bearer " + session.token);
```

Comment:
A partir do momento que você utiliza a rota testada mais de uma vez no próprio `it()` você acaba criando uma dependência de assertividade em uma rota ainda não testada.